tools/ctl/ide/ui.go

```diff
@@ -216,6 +227,78 @@ func (e *explorer) showLogs(ctx context.Context, example firestore.Rebuild) {
 	}
 }
 
+func (e *explorer) editAndRun(ctx context.Context, example firestore.Rebuild) {
+	localAssets, err := localAssetStore(ctx, example.Run)
+	if err != nil {
+		log.Println(errors.Wrap(err, "failed to create local asset store"))
+		return
+	}
+	buildDefAsset := rebuild.Asset{Type: rebuild.BuildDef, Target: example.Target()}
+	var currentStratYaml []byte
+	{
+		if r, _, err := localAssets.Reader(ctx, buildDefAsset); err == nil {
+			buf := new(bytes.Buffer)
+			_, err = io.Copy(buf, r)
+			if err != nil {
+				log.Println(errors.Wrap(err, "failed to read existing build definition"))
+				return
+			}
+			currentStratYaml = buf.Bytes()
+		} else {
+			var oneof schema.StrategyOneOf
+			if err := json.Unmarshal([]byte(example.Strategy), &oneof); err != nil {
+				log.Println(errors.Wrap(err, "failed to parse strategy"))
+				return
+			}
+			var err error
+			if currentStratYaml, err = yaml.Marshal(oneof); err != nil {
+				log.Println(errors.Wrap(err, "failed to marshal strategy"))
+				return
+			}
+		}
+	}
+	var newStratYaml []byte
```

Comment:
and for parity, I'd also make this a oneof so this "solicit input from user" step can fail atomically if the user inputs an invalid strat.